SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/VisibilityTest.cpp

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "VisibilityTest.h"
+#include "ReplicatedVisibilityTestActor.h"
+#include "GameFramework/PlayerController.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMovement/TestMovementCharacter.h"
+#include "GameFramework/CharacterMovementComponent.h"
+#include "Kismet/GameplayStatics.h"
+#include "Net/UnrealNetwork.h"
+#include "Containers/Array.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKSettings.h"
+
+/**
+ * This test tests if a bHidden Actor is replicating properly to Server and Clients.
+ * 
+ * The test includes a single server and two client workers.
+ * The flow is as follows:
+ *  - Setup:
+ *	  - On cube actor already placed in the level Location FVector(0.0f, 0.0f, 50.0f)
+ *    - Two test character pawns are spawned, one for each client
+ *    - The controllers for each client  possess the spawned test character pawns
+ *  - Test:
+ *    - Server moves one Client to a remote position where Client can not see the HiddenActor(Cube) .
+ *	  - Check if the Clients can see the HiddenActor(Cube).
+ *	  - Server sets the HiddenActor(Cube) as bHidden.
+ *	  - Server moves the remote Client back close to the HiddenActor(Cube).
+ *	  - Check if the Clients can see the HiddenActor(Cube).
+ *	  - Server sets the HiddenActor(Cube) as Visible (bHidden = false).
+ *	  - Check if the Clients can see the HiddenActor(Cube).
+ *  - Cleanup:
+ *    - The clients repossess their default pawns
+ *    - The test pawns are destroyed
+ */
+
+AVisibilityTest::AVisibilityTest()
+	: Super()
+{
+	Author = "Evi";
+	Description = TEXT("Test Actor Visibility");
+}
+void AVisibilityTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(AVisibilityTest, Controllers);
+	DOREPLIFETIME(AVisibilityTest, TestPawns);
+}
+
+void AVisibilityTest::BeginPlay()
+{
+	Super::BeginPlay();
+	PreviousPositionUpdateFrequency = GetDefault<USpatialGDKSettings>()->PositionUpdateFrequency;
+
+	CharacterRemoteLocation = FVector(20000.0f, 20000.0f, 50.0f);
+	Character1StartingLocation = FVector(0.0f, 120.0f, 50.0f);
+	Character2StartingLocation = FVector(0.0f, 240.0f, 50.0f);
```

Comment:
I would prefer writing the test in such a way that it can have any number of clients.
E.g. 120 + 120*numClients here instead.